diff --git a/libpointmatcher_ros/CMakeLists.txt b/libpointmatcher_ros/CMakeLists.txt
index a29df59..b92f0d9 100644
--- a/libpointmatcher_ros/CMakeLists.txt
+++ b/libpointmatcher_ros/CMakeLists.txt
@@ -9,7 +9,6 @@ nav_msgs
 tf
 tf_conversions
 eigen_conversions
-libpointmatcher
 )
 
 catkin_package(
@@ -22,9 +21,10 @@ nav_msgs
 tf
 tf_conversions
 eigen_conversions
-libpointmatcher
 )
 
+find_package(libpointmatcher REQUIRED)
+
 # c++ 0x
 if("${CMAKE_CXX_COMPILER_ID}" MATCHES "[cC]lang")
 	set(CMAKE_COMPILER_IS_CLANGXX)
@@ -48,11 +48,11 @@ set(EXECUTABLE_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/bin)
 set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
 
 add_definitions(-DHAVE_YAML_CPP)
-include_directories(include)
+include_directories(include ${libpointmatcher_INCLUDE_DIRS})
 
 add_library(pointmatcher_ros src/point_cloud.cpp src/transform.cpp)
 include_directories(${catkin_INCLUDE_DIRS})
-target_link_libraries(pointmatcher_ros ${catkin_LIBRARIES})
+target_link_libraries(pointmatcher_ros ${catkin_LIBRARIES} ${libpointmatcher_LIBRARIES})
 
 # Mark executables and/or libraries for installation
 install(TARGETS pointmatcher_ros
